judge/views/problem.py

```diff
@@ -461,6 +462,50 @@ def post(self, request, *args, **kwargs):
         return HttpResponseRedirect(request.get_full_path())
 
 
+class SuggestList(ProblemList):
+    template_name = 'problem/suggest-list.html'
+    permission_required = "superuser"
+
+    def get_normal_queryset(self):
+        filter = ~Q(suggesters=None) & Q(is_public=False)
+        if self.profile is not None:
+            filter |= Q(authors=self.profile)
+            filter |= Q(curators=self.profile)
+            filter |= Q(testers=self.profile)
+        queryset = Problem.objects.filter(filter).select_related('group').defer('description', 'summary')
+        if not self.request.user.has_perm('see_organization_problem'):
+            filter = Q(is_organization_private=False)
+            if self.profile is not None:
+                filter |= Q(organizations__in=self.profile.organizations.all())
+            queryset = queryset.filter(filter)
+        if self.show_types:
+            queryset = queryset.prefetch_related('types')
+        if self.category is not None:
+            queryset = queryset.filter(group__id=self.category)
+        if self.selected_types:
+            queryset = queryset.filter(types__in=self.selected_types)
+        if 'search' in self.request.GET:
+            self.search_query = query = ' '.join(self.request.GET.getlist('search')).strip()
+            if query:
+                if settings.ENABLE_FTS and self.full_text:
+                    queryset = queryset.search(query, queryset.BOOLEAN).extra(order_by=['-relevance'])
+                else:
+                    queryset = queryset.filter(
+                        Q(code__icontains=query) | Q(name__icontains=query) | Q(source__icontains=query) |
+                        Q(translations__name__icontains=query, translations__language=self.request.LANGUAGE_CODE))
+        self.prepoint_queryset = queryset
+        if self.point_start is not None:
+            queryset = queryset.filter(points__gte=self.point_start)
+        if self.point_end is not None:
+            queryset = queryset.filter(points__lte=self.point_end)
+        return queryset.distinct()
+
+    def get(self, request, *args, **kwargs):
+        if not request.user.is_superuser and not request.user.is_staff:
```

Comment:
`not request.user.is_staff` -> `not request.user.has_perm('suggest_new_problem')`